### PR TITLE
Upgrade Navigator to Webpack 5

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: navigator/frontend/webpack.config.js [hunk 1]
```diff
@@ -85,15 +85,15 @@ module.exports = (env) => {
   }
 
   return {
-    mode: isProduction ? 'production' : 'development',
+    mode: isProduction ? 'development' : 'development',
```
[the remark this answers is not in the document]
Comment (cocreature) [reply navigator/frontend/webpack.config.js:88]: oh definitely, I was hoping to get something more sensible for debugging out of this. I didn’t. `production` produces the same error.

File: navigator/frontend/src/browsercheck.ts [hunk 2]
```diff
@@ -37,6 +37,10 @@ const features = [
 
 // Find all features from above list that are not supported in current browser
 const missingFeatures = [];
+console.log(typeof Modernizr);
+console.log(Object.keys(Modernizr));
+console.log(Modernizr);
+console.log(Modernizr["es5undefined"]);
```
Comment (stefanobaghino-da) [inline navigator/frontend/src/browsercheck.ts:43]: I guess these are debug prints.

Comment (stefanobaghino-da) [inline navigator/frontend/src/browsercheck.ts:43]: I guess these are debug prints.

Comment (cocreature) [reply navigator/frontend/src/browsercheck.ts:43]: yeah will clean it up once I figured out modernizr

Comment (stefanobaghino-da) [inline navigator/frontend/src/browsercheck.ts:43]: I guess these are debug prints.

**Review by stefanobaghino-da — COMMENTED** [review]
I went through some of these changes in my first attempt. Why do you need `modernizr` though?

**Comment by cocreature** [comment]
> I went through some of these changes in my first attempt. Why do you need `modernizr` though?

Not sure we need it but we used it before so it seemed broadly sensible to keep it. I’ve now pushed a change that disables the checks that are broken in strict mode which fixes it. They’re not particularly useful anyway so we might just want to kill modernizr completely but probably better in a separate PR.

**Review by stefanobaghino-da — APPROVED** [review]

**Pull request merged**